### ISSUE TITLE
vmci support

### ISSUE THREAD

**Issue by rudgab** [zeromq/libzmq#1655]
vmci_sockets.h is missing

make[1]: Entering directory '/tmp/libzmq'
  CXX      src/src_libzmq_la-address.lo
In file included from src/address.cpp:40:0:
src/vmci_address.hpp:39:26: fatal error: vmci_sockets.h: No such file or directory
 #include <vmci_sockets.h>
                          ^
compilation terminated.
Makefile:2984: recipe for target 'src/src_libzmq_la-address.lo' failed
make[1]: *** [src/src_libzmq_la-address.lo] Error 1
make[1]: Leaving directory '/tmp/libzmq'
Makefile:3682: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

**Comment by kevinkreiser** [comment]
So far I've tested on ubuntu 12.04 and 14.04 and when configured like so:

./configure --without-libsodium --without-documentation

They both do this:

In file included from src/socket_base.cpp:73:0:
src/vmci_address.hpp:39:26: fatal error: vmci_sockets.h: No such file or directory
 #include <vmci_sockets.h>
                          ^

Interestingly configure seems to say that by default its disabled:

  --with-vmci             build libzmq with VMCI transport [default=no]

And indeed I dont see anything in the Makefile directly setting ZMQ_HAVE_VMCI but its generated and a bit hard to follow. Any clue what is going on here? Note that sending --with-vmci=no in configure does work though.

This might suggest that there is a problem with the default as well as the detection in configure.ac or other autotools stuff.

**Comment by bluca** [comment]
There's very 2 small issues, very easy to solve, opened a PR.
